components/raftstore/src/coprocessor/mod.rs

```diff
@@ -219,33 +219,59 @@ impl ObserveHandle {
     }
 }
 
+// `ObserveLevel` describe what data the observer want to observe
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ObserveLevel {
+    // Don't observe any data
+    None,
+    // Only observe lock related data (i.e `lock_cf`, `write_cf`)
+    LockRelated,
+    // Observe all data
+    All,
+}
+
+impl ObserveLevel {
+    fn from(cdc: &ObserveHandle, rts: &ObserveHandle) -> ObserveLevel {
+        match (cdc.is_observing(), rts.is_observing()) {
```

Comment:
It's quite misleading and easy to write the wrong code. How about let it return `ObserveLevel` directly and use the maximum one?